### ISSUE TITLE
Traverse should stop at the first Error value.

### ISSUE THREAD

**Issue by pangwa** [fsprojects/FSharpPlus#415]
### Description

It should not evaluate list after the first error happend.

### Repro steps

Below code could reproduce the problem:
```f#
let mapper v = 
    Console.WriteLine (sprintf "mapping %d " v)
    if v > 3 then Result.Error(sprintf "error: %d" v)
    else Result.Ok v

let testTrav l = 
    let v = Seq.traverse mapper l 
    v |> Result.map Seq.last

let r = Seq.ofList [1; 2; 3; 4; 5] |> testTrav
match r with
| Result.Ok v -> Console.WriteLine (sprintf "got %d" v)
| Result.Error err -> Console.WriteLine (sprintf "failed %s" err)
```

### Expected behavior
It should prints:

> mapping 1
> mapping 2
> mapping 3
> mapping 4
> failed error: 4

### Actual behavior
The output is:
> mapping 1
> mapping 2
> mapping 3
> mapping 4
> mapping 5
> failed error: 4

### Related information 
Tested with v1.1.6

**Comment by pangwa** [comment]
Closing it as it's been fixed by #416

**Comment by wallymathieu** [comment]
Thanks for finding it @pangwa !
